### PR TITLE
feat: Feature/intmbd 1125 sagemaker quickstart

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Terraform module for creating sagemake quickstart.

feat: Feature/intmbd 1125 sagemaker quickstart

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

**Review by marcosuma — APPROVED** [review]
LGTM - thanks for the demo shown us last Friday!

File: examples/Atlas_Database_Users/atlas_cluster.tf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 resource "mongodbatlas_cluster" "cluster" {
   project_id             = mongodbatlas_project.project1.id
-  name                   = "MongoDB_Atlas"
+  name                   = "MongoDB-Atlas-cluster-tf-test"
```
Comment (maastha) [inline examples/Atlas_Database_Users/atlas_cluster.tf]: [q] why do we need to change this?

File: modules/atlas-basic/versions.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_providers {
+    mongodbatlas = {
+      source  = "mongodb/mongodbatlas"
+      version = "0.9.1"
```
Comment (maastha) [inline modules/atlas-basic/versions.tf]: this is a very old version, latest is 1.12.1

File: modules/examples/sagemaker/outputs.tf
Comment (maastha) [inline modules/examples/sagemaker/outputs.tf:1]: needed?

File: modules/atlas-basic/aws-vpc.tf [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  description = "Default security group for all instances in vpc"
+  vpc_id      = aws_vpc.vpc_east.id
+  ingress {
+    from_port = 0
+    to_port   = 0
+    protocol  = "tcp"
+    cidr_blocks = [
+      "0.0.0.0/0",
+    ]
+  }
```
Comment (andreaangiolillo) [inline modules/atlas-basic/aws-vpc.tf]: this is not a best practise. Let's make this configurable and provide a cidr_block var

File: modules/atlas-basic/aws-vpc.tf [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (andreaangiolillo) [inline modules/atlas-basic/aws-vpc.tf]: same, using `0.0.0.0/0` is not a best practise and we should not include this in the module

File: modules/atlas-basic/variables.tf [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+variable "cidr_block" {
+    description = "IP Access List CIDRs"
+    type        = list(string)
+    default     = ["10.1.0.0/16","12.2.0.0/16"]
+}
+
+variable "ip_address" {
+    description = "IP Access List IP Addresses"
+    type        = list(string)
+    default     = ["208.169.90.207","0.0.0.0"]
```
Comment (andreaangiolillo) [inline modules/atlas-basic/variables.tf]: we should not provide default here as we don't now the IP addressed of the machines that will access the database
```suggestion
```

File: modules/atlas-basic/variables.tf [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+variable "aws_vpc_cidr_block" {
+    description = "AWS VPC CIDR block"
+    default     = "10.0.0.0/16"
+    type        = string
+
+}
+
+variable "aws_route_table_cidr_block" {
+    description = "AWS route table CIDR block"
+    default     = "0.0.0.0/0"
```
Comment (andreaangiolillo) [inline modules/atlas-basic/variables.tf:154]: lest remove `"0.0.0.0/0"` from everywhere 
```suggestion
```

File: modules/atlas-basic/variables.tf [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+variable "aws_subnet_availability_zone1" {
+    description = "AWS subnet availability zone"
+    default     = "us-east-1a"
+    type        = string
+}
+variable "aws_subnet_availability_zone2" {
+  description = "AWS subnet availability zone"
+  default     = "us-east-1b"
```
Comment (andreaangiolillo) [inline modules/atlas-basic/variables.tf:174]: let;s use us-east-1 instead

File: modules/atlas-basic/versions.tf (hunk 2 above)
Comment (andreaangiolillo) [reply modules/atlas-basic/versions.tf]: +1

File: modules/examples/atlas-basic/mian.tf [hunk 8]
```diff
@@ -0,0 +1,10 @@
+module "atlas-basic"  {
+  source = "../../atlas-basic"
```
Comment (andreaangiolillo) [inline modules/examples/atlas-basic/mian.tf]: the file name should be `main.tf`

File: modules/examples/sagemaker/sagemaker-example/lambda_functions/process_mdb_change_event/app.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
+import boto3
+import json
```
Comment (andreaangiolillo) [inline modules/examples/sagemaker/sagemaker-example/lambda_functions/process_mdb_change_event/app.py]: let's remove the sagemake example code from this repo and link to the code in the cloud-formation for this https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples/quickstart-mongodb-atlas-analytics-amazon-sagemaker-integration

File: modules/mongodb-atlas-analytics-amazon-sagemaker-integration/versions.tf [hunk 10]
```diff
@@ -0,0 +1,13 @@
+terraform {
+  required_providers {
+    mongodbatlas = {
+      source  = "mongodb/mongodbatlas"
+      version = "1.12.0"
```
Comment (andreaangiolillo) [inline modules/mongodb-atlas-analytics-amazon-sagemaker-integration/versions.tf]: this is not the latest

**Review by andreaangiolillo — CHANGES_REQUESTED** [review]

File: modules/atlas-basic/versions.tf (hunk 2 above)
Comment (govardhanpagidi) [reply modules/atlas-basic/versions.tf]: Updated

File: examples/Atlas_Database_Users/atlas_cluster.tf (hunk 1 above)
Comment (govardhanpagidi) [reply examples/Atlas_Database_Users/atlas_cluster.tf]: Reverted

File: modules/atlas-basic/aws-vpc.tf (hunk 3 above)
Comment (govardhanpagidi) [reply modules/atlas-basic/aws-vpc.tf]: Reverted

File: modules/atlas-basic/aws-vpc.tf (hunk 4 above)
Comment (govardhanpagidi) [reply modules/atlas-basic/aws-vpc.tf]: Updated

File: modules/atlas-basic/variables.tf (hunk 5 above)
Comment (govardhanpagidi) [reply modules/atlas-basic/variables.tf]: Reverted

File: modules/atlas-basic/variables.tf (hunk 6 above)
Comment (govardhanpagidi) [reply modules/atlas-basic/variables.tf:154]: Reverted

File: modules/examples/atlas-basic/mian.tf (hunk 8 above)
Comment (govardhanpagidi) [reply modules/examples/atlas-basic/mian.tf]: Updated

File: modules/examples/sagemaker/sagemaker-example/lambda_functions/process_mdb_change_event/app.py (hunk 9 above)
Comment (govardhanpagidi) [reply modules/examples/sagemaker/sagemaker-example/lambda_functions/process_mdb_change_event/app.py]: Removed

File: modules/atlas-basic/variables.tf (hunk 7 above)
Comment (maastha) [reply modules/atlas-basic/variables.tf:174]: availability zones are supposed to be 'us-east-1a',  'us-east-1b', etc

**Review by maastha — APPROVED** [review]
LGTM

File: modules/atlas-basic/variables.tf (hunk 6 above)
Comment (andreaangiolillo) [reply modules/atlas-basic/variables.tf:154]: I still see `default = "0.0.0.0/0"`
Am I missing something?

File: examples/test-upgrade/v090/cluster/v082/main.tf [hunk 11]
```diff
@@ -9,7 +9,6 @@ resource "mongodbatlas_cluster" "test" {
   disk_size_gb                 = 100
   num_shards                   = 1
   replication_factor           = 3
-  provider_backup_enabled      = true
```
Comment (andreaangiolillo) [inline examples/test-upgrade/v090/cluster/v082/main.tf:12]: why do we need this change?  this is an example of upgrading the provider from v082 to 090

File: examples/test-upgrade/v090/cluster/v090/main.tf [hunk 12]
```diff
@@ -9,7 +9,6 @@ resource "mongodbatlas_cluster" "test" {
   disk_size_gb                 = 100
   num_shards                   = 1
   replication_factor           = 3
-  provider_backup_enabled      = true
```
Comment (andreaangiolillo) [inline examples/test-upgrade/v090/cluster/v090/main.tf:12]: why is this needed?

Comment (govardhanpagidi) [reply examples/test-upgrade/v090/cluster/v090/main.tf:12]: this property is not expected in cluster, thrown by the [workflow here](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/6314926951/job/17146303344#step:11:2829)

File: examples/test-upgrade/v090/cluster/v082/main.tf (hunk 11 above)
Comment (govardhanpagidi) [reply examples/test-upgrade/v090/cluster/v082/main.tf:12]: this property is not expected in cluster, thrown by the [workflow here](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/6314926951/job/17146303344#step:11:2829)

**Comment by govardhanpagidi** [comment]
@andreaangiolillo "0.0.0.0/0" is  expected  for IGW [here](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1494#discussion_r1340122244)

**Review by andreaangiolillo — APPROVED** [review]
